lib/core/src/display/shape/primitive/glsl/color.glsl

```diff
@@ -189,28 +235,164 @@ HSV hsv(RGB rgb) {
+    if (offset < gradient.control_point1.offset) {                                                 \
+        return gradient.control_point1.color;                                                      \
+    } else if (offset < gradient.control_point2.offset) {                                          \
+        return sample(GRADIENT2(gradient.control_point1,gradient.control_point2),offset);          \
+    } else if (offset < gradient.control_point3.offset) {                                          \
+        return sample(GRADIENT2(gradient.control_point2,gradient.control_point3),offset);          \
+    } else if (offset < gradient.control_point4.offset) {                                          \
+        return sample(GRADIENT2(gradient.control_point3,gradient.control_point4),offset);          \
+    } else {                                                                                       \
+        return gradient.control_point4.color;                                                      \
+    }                                                                                              \
+}                                                                                                  \
+                                                                                                   \
+COLOR sample(GRADIENT5 gradient, float offset) {                                                   \
+    if (offset < gradient.control_point1.offset) {                                                 \
+        return gradient.control_point1.color;                                                      \
+    } else if (offset < gradient.control_point2.offset) {                                          \
+        return sample(GRADIENT2(gradient.control_point1,gradient.control_point2),offset);          \
+    } else if (offset < gradient.control_point3.offset) {                                          \
+        return sample(GRADIENT2(gradient.control_point2,gradient.control_point3),offset);          \
+    } else if (offset < gradient.control_point4.offset) {                                          \
+        return sample(GRADIENT2(gradient.control_point3,gradient.control_point4),offset);          \
+    } else if (offset < gradient.control_point5.offset) {                                          \
+        return sample(GRADIENT2(gradient.control_point4,gradient.control_point5),offset);          \
+    } else {                                                                                       \
+        return gradient.control_point5.color;                                                      \
+    }                                                                                              \
+}                                                                                                  \
+
+
+DEF_GRADIENT(Rgb,rgb,RgbGradientControlPoint,RgbGradient1,RgbGradient2,RgbGradient3,RgbGradient4,RgbGradient5)
```

Comment:
In c you can do concatenating identifiers in macros like"
`struct GRADIENT ## 1`
So you could have then
`DEF_GRADIENT(Rgb,rgb,RgbGradientControlPoint,RgbGradient)`
Maybe glsl also have such utility?

Reply:
It has, but not in WebGL -.-